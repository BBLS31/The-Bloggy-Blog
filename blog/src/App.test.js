import React from "react";
import { render, waitFor, screen } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import Main from "../src/components/pages/blogpages/main/main";
import "isomorphic-fetch";




test("Main component renders correctly", () => {
  const { asFragment } = render(
    <Router>
      <Main />
    </Router>
  );
  expect(asFragment()).toMatchSnapshot();
});




test('Main component fetches and renders a blog post', async () => {
  const { getByText } = render(
    <Router>
      <Main />
    </Router>
  );

  // Wait for the fetch request to complete and the blog post to be rendered
  await waitFor(() => getByText('Being Him'))
});
