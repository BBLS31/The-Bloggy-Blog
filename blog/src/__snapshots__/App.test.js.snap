// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="without-user"
      >
        <a
          class="nav-brand"
          href="/"
        >
          <h2>
            The Bloggy Blog
          </h2>
        </a>
        <div
          class="btns"
        >
          <button
            class="home-link-btn"
          >
            <a
              class="home-link"
              href="/"
            >
               Home
            </a>
          </button>
          <button
            class="s-btn"
          >
            <a
              class="signup-link"
              href="/signup"
            >
              Signup
            </a>
          </button>
          <button
            class="lg-btn"
          >
            <a
              class="lg-link"
              href="/login"
            >
              Login
            </a>
          </button>
        </div>
      </div>
    </div>
    <h1
      class="page-title"
    >
      The Posts
    </h1>
  </div>
</DocumentFragment>
`;

exports[`Main renders correctly 1`] = `
<DocumentFragment>
  <div
    class="main"
  >
    <div
      class="header"
    >
      <div
        class="without-user"
      >
        <a
          class="nav-brand"
          href="/"
        >
          <h2>
            The Bloggy Blog
          </h2>
        </a>
        <div
          class="btns"
        >
          <button
            class="home-link-btn"
          >
            <a
              class="home-link"
              href="/"
            >
               Home
            </a>
          </button>
          <button
            class="s-btn"
          >
            <a
              class="signup-link"
              href="/signup"
            >
              Signup
            </a>
          </button>
          <button
            class="lg-btn"
          >
            <a
              class="lg-link"
              href="/login"
            >
              Login
            </a>
          </button>
        </div>
      </div>
    </div>
    <h1
      class="page-title"
    >
      The Posts
    </h1>
  </div>
</DocumentFragment>
`;
